### ISSUE TITLE
sql: remove enforce_home_region_follower_reads_enabled

### ISSUE THREAD

**Issue by miraradeva** [cockroachdb/cockroach#113765]
Currently, the `enforce_home_region_follower_reads_enabled` feature uses (1) savepoint rollback and (2) follower reads to dynamically detect a home region for a query which encountered "Query has no home region" error. (1) is used to roll back the original failed read, and (2) is used to ensure that while we try different regions to find the home region we don't actually serve remote reads.

In #111228 we want to change the behavior of savepoint rollbacks to not drop the accumulated read spans, and instead ensure those reads are serializable even after the rollback. This change breaks the above `enforce_home_region_follower_reads_enabled`  functionality because (1) no longer removes the original failed read from the txn's read span, and as a result (2) fails to use AOST because the txn has preceding reads.

Ideally, we would change the behavior of `enforce_home_region_follower_reads_enabled` to work with the new savepoint behavior but it's unclear how to best approach this. In the mean time, after discussing with @michae2, we agreed we can limit the behavior of `enforce_home_region_follower_reads_enabled`  to work only for queries that already use AOST. This will help because we no longer need to do part (2). 

Jira issue: CRDB-33142

**Comment by mgartner** [comment]
@michae2 Now that #129024 deprecates this setting, is there any more work here planned, besides the eventual removal of the setting? We're no longer planning on limiting it to AOST queries as the title suggests, correct?

**Comment by michae2** [comment]
> @michae2 Now that #129024 deprecates this setting, is there any more work here planned, besides the eventual removal of the setting? We're no longer planning on limiting it to AOST queries as the title suggests, correct?

Correct, now planning to remove `enforce_home_region_follower_reads_enabled`, leaving just the `enforce_home_region` behavior. I can change the title to that.
